### ISSUE TITLE
Theme demo broken on the Hugo Themes website

### ISSUE THREAD

**Issue by onedrawingperday** [marketempower/axiom#5]
Hello @jhauraw 

Your recent theme updates broke the theme demo on the Hugo website.

The reason for the breakage is that the architecture of the Themes Site is different than the average Hugo project and that is why in the README we ask theme authors not to harcdcode PATHs.

In the `baseof.html` template you are making a call to `package.json` that lives under the root of your theme's repo to fetch some meta data about the theme's name and version.

However to make Hugo find `package.json` you need to change [lines 12-15](https://github.com/marketempower/axiom/blob/master/layouts/_default/baseof.html#L12-L15) of `default/baseof.html` to the following:

```
  {{- if ne .Site.Params.omitThemeAttribution true -}}
  {{- $pkgJson := . -}}
    <!-- Check for the Hugo Themes demo because the PATH to package.json is different -->
  {{- if eq .Site.Title "Hugo Themes" -}}
  {{- $pkgJson = getJSON "../../../axiom/package.json" -}}
  {{- else -}}
  {{- $pkgJson = getJSON "./themes/axiom/package.json" -}}
  {{- end }}
  <meta name="hugo-theme" content="{{ printf "%s %s" $pkgJson.name $pkgJson.version | safeHTMLAttr }}">
  {{- end }}
```
Basically in the above a check is made whether the title of the demo is "Hugo Themes" -that is the default title served by the config for theme demos that do not provide an exampleSite, like yours- and a different relative PATH is specified for `package.json` so that it is found.

Also note that I have removed your construct `(T "theme" .)` I don't quite understand what you are trying to render with it, it doesn't work and I've replaced it with `$pkgJson.name` to print the theme name from `package.json`. I also noticed that you are using this construct in other templates and it doesn't output anything so I suggest that you remove it.

Furthermore I noticed that you are using some obsolete meta tags `<meta name="robots" content="noodp,noydir">` These are no longer needed and you should remove them, see [here](https://searchfacts.com/noodp-noydir-meta-tags/) for all the details about these tags.

Once you update the theme to fix the broken demo feel free to close this issue.

cc: @digitalcraftsman

**Comment by jhauraw** [comment]
@onedrawingperday hello, thank you for taking the time to alert me of this issue.

Is there a better/cleaner way to identify if it's the Themes demo site than using `.Site.Title`? I was thinking the Config file `baseurl`? Although I really don't like the idea of an if/else just for the themes site.

Alternatively, a more universal approach may be to use an absolute URL to the packege.json file on Github? With this method we don't need the if/else to check if it's the Hugo Themes site. Thoughts?

**Comment by onedrawingperday** [comment]
> Is there a better/cleaner way to identify if it's the Themes demo site than using .Site.Title? I was thinking the Config file baseurl? Although I really don't like the idea of an if/else just for the themes site.

There is no Environment variable for Hugo Themes that I am aware of. What I posted above has been tested with the Hugo Themes Build Script. I tried variations with the `baseURL` but I did not get it working.

>Alternatively, a more universal approach may be to use an absolute URL to the packege.json file on Github? With this method we don't need the if/else to check if it's the Hugo Themes site. Thoughts?

If you prefer  this approach then go ahead. `getJSON` can fetch an external JSON.

**Comment by jhauraw** [comment]
@onedrawingperday the new v0.6.7 release of Axiom resolves all the issues reported. I found a very clean way to deal with the getJSON path issue.

I tested against the `hugoThemes` repo and ran the `./generateThemeSite.sh` build process, and a Demo site is generated without errors.

Thanks again for the heads up. Once I get confirmation from your end that the build process completes, I'll close this issue.
